tools/ui-components/src/form-group/form-group.stories.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Story } from '@storybook/react';
+import { Button } from '../button';
+import { FormControl } from '../form-control';
+
+import { FormGroup, FormGroupProps } from '.';
+
+const story = {
+  title: 'Example/FormGroup',
+  component: FormGroup,
+  argTypes: {
+    children: { control: { type: 'object' } },
+    className: { control: { type: 'text' } },
+    controlId: { control: { type: 'text' } },
+    validationState: { options: ['success', 'warning', 'error'] }
+  }
+};
+
+const Template: Story<FormGroupProps> = ({
+  controlId,
+  className,
+  validationState
+}) => {
+  return (
+    <div>
+      {/* when a function runs and confirm a value it should push vaidation state into formGroup */}
+      <Button onClick={() => (validationState = 'success')}>Success</Button>
+      <Button onClick={() => (validationState = 'warning')}>Warning</Button>
+      <Button onClick={() => (validationState = 'error')}>Error</Button>
+      <FormGroup
+        controlId={controlId}
+        className={className}
+        validationState={validationState}
+      >
+        <FormControl
+          type={'text'}
+          placeholder='Hello World'
+          defaultValue='Hello World'
```

Comment:
if this changed to `value` like `value='Hello World'`, we will get error in the console.

![image](https://user-images.githubusercontent.com/88248797/200156248-d4b6662d-917f-42af-a6b0-8c7df685220f.png)


Is it possible to add type check, that warn the developer when they add value without `onChange` or `readOnly` props in the FormControl?

Reply:
Typescript conditional types might come handy: https://www.typescriptlang.org/docs/handbook/2/conditional-types.html

Reply:
The fix is in #48696 👍